client/question/question.js

```diff
@@ -550,101 +550,195 @@ tie.constant(
   'PREREQ_CHECK_TYPE_INVALID_STUDENTCODE_CALL', 'invalidStudentCode');
 
 /**
-  Dictionary of the customized runtime error feedback messages
-  - `checker` function checks to see if the error matches with the type of
-    feedback message
-  - `generateMessage` function returns the appropriate feedback message for the
-    given error
+ * Dictionary of the novice-friendly syntax error feedback messages
+ *
+ * `friendlyErrorCheck` function checks to see if the error matches with the
+ * type of feedback message
+ *
+ * `getFriendlyErrorText` function returns the appropriate feedback message for
+ * the given error
+ *
+ * @type {Object}
+ */
+tie.constant('FRIENDLY_SYNTAX_ERROR_TRANSLATIONS', {
+  python: [
+    {
+      friendlyErrorCheck: function(errorString) {
+        var re = new RegExp('^NameError: global name .* is not defined$');
+        return re.test(errorString);
+      },
+      getFriendlyErrorText: function(errorString) {
+        var re = new RegExp('NameError: global name (.*) is not defined');
+        var varName = errorString.match(re)[1];
+        return [
+          '<code>', varName, '</code> appears to be a name that was not ',
+          'previously declared or defined. E.g., if <code>', varName,
```

Comment:
(Optional) This seems a little verbose IMO -- the most likely issue here is a typo of the method/function/variable/etc. name. What do you think of something shorter, like:

```
    It looks like varName wasn't previously declared or defined. Did you type it
    incorrectly, here or in the rest of the code? (Note that using consistent
    capitalization is also important.)
```

That said, please feel free to ignore this suggestion if you prefer the original.

Reply:
It might be verbose to someone familiar with Python and coding, yes. I think it's a good opportunity to remind the novice how to "declare" a variable.

Reply:
I'm happy to defer here, per offline conversation.